### PR TITLE
docs: add code snippets for llm text generatiion

### PR DESCRIPTION
**Pull request by ashleyxuu**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes internal #338873779 🦕

### CODE REVIEW

File: samples/snippets/text_generation_test.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    # [START bigquery_dataframes_perform_keyword_extraction]
+    import bigframes.pandas as bpd
+
+    df = bpd.read_gbq("bigquery-public-data.imdb.reviews").head(n=5)
+    df_prompt_prefix = "Extract the key words from the text below: "
+    df_prompt = df_prompt_prefix + df["review"]
+
+    # Predict using the model
+    df_pred = model.predict(
+        df_prompt.to_frame(), temperature=0.2, max_output_tokens=100
```
Comment (shobsi) [inline samples/snippets/text_generation_test.py:44]: `.to_frame()` seems redundant, the method accepts `Union[bpd.DataFrame, bpd.Series]`. Same at line 60

Comment (ashleyxuu) [reply samples/snippets/text_generation_test.py]: Thanks. Addressed.

File: samples/snippets/text_generation_test.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    LOCATION = "us"
+
+    # [START bigquery_dataframes_create_remote_model]
+    import bigframes
+    from bigframes.ml.llm import PaLM2TextGenerator
+
+    bigframes.options.bigquery.project = PROJECT_ID
+    bigframes.options.bigquery.location = LOCATION
+
+    model = PaLM2TextGenerator()
```
Comment (shobsi) [inline samples/snippets/text_generation_test.py:31]: GeminiTextGenerator would be a more promising showcase. Do we plan to add that too in a separate PR?

**Review by shobsi — APPROVED** [review]

File: samples/snippets/text_generation_test.py (hunk 2 above)
Comment (ashleyxuu) [reply samples/snippets/text_generation_test.py:31]: This is for code snippets in tutorials here: https://cloud.google.com/bigquery/docs/generate-text-tutorial#create_the_remote_model. 

We need to be consistent with the tutorials using the text-bison model

Comment (shobsi) [reply samples/snippets/text_generation_test.py:31]: Got it, thank you!

File: samples/snippets/text_generation_test.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    import bigframes.pandas as bpd
+
+    df = bpd.read_gbq("bigquery-public-data.imdb.reviews").head(n=5)
+    df_prompt_prefix = "Extract the key words from the text below: "
+    df_prompt = df_prompt_prefix + df["review"]
+
+    # Predict using the model
+    df_pred = model.predict(df_prompt, temperature=0.2, max_output_tokens=100)
+    df_pred.peek(5)
+    # [END bigquery_dataframes_perform_keyword_extraction]
```
Comment (tswast) [inline samples/snippets/text_generation_test.py:45]: Let's add a comment here for our own sake reminding us to update screenshots if we change this sample.

```suggestion
    # [END bigquery_dataframes_perform_keyword_extraction]
    # peek() is used to show a preview of the results. If the output
    # of this sample changes, also update the screenshot for the associated
    # tutorial on cloud.google.com.
```

File: samples/snippets/text_generation_test.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    import bigframes.pandas as bpd
+
+    df = bpd.read_gbq("bigquery-public-data.imdb.reviews").head(n=5)
+    df_prompt_prefix = "perform sentiment analysis on the following text, return one the following categories: positive, negative: "
+    df_prompt = df_prompt_prefix + df["review"]
+
+    # Predict using the model
+    df_pred = model.predict(df_prompt, temperature=0.2, max_output_tokens=100)
+    df_pred.peek(5)
+    # [END bigquery_dataframes_perform_sentiment_analysis]
```
Comment (tswast) [inline samples/snippets/text_generation_test.py:59]: Same comment here as a reminder to us.

```suggestion
    # [END bigquery_dataframes_perform_sentiment_analysis]
    # peek() is used to show a preview of the results. If the output
    # of this sample changes, also update the screenshot for the associated
    # tutorial on cloud.google.com.
```

File: samples/snippets/text_generation_test.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    df_pred = model.predict(df_prompt, temperature=0.2, max_output_tokens=100)
+    df_pred.peek(5)
+    # [END bigquery_dataframes_perform_keyword_extraction]
+    assert df_pred["ml_generate_text_llm_result"] is not None
+    assert df_pred["ml_generate_text_llm_result"].iloc[0] is not None
+
+    # [START bigquery_dataframes_perform_sentiment_analysis]
+    import bigframes.pandas as bpd
+
+    df = bpd.read_gbq("bigquery-public-data.imdb.reviews").head(n=5)
```
Comment (tswast) [inline samples/snippets/text_generation_test.py:52]:
```suggestion
    df = bpd.read_gbq("bigquery-public-data.imdb.reviews", max_results=5)
```

**Review by tswast — CHANGES_REQUESTED** [review]
Thanks! A few nitpicks.

File: samples/snippets/text_generation_test.py (hunk 4 above)
Comment (ashleyxuu) [reply samples/snippets/text_generation_test.py]: Got it. Thanks

**Review by tswast — APPROVED** [review]
Thanks!

**Comment by tswast** [comment]
I've mailed internal CL 632203837 to address the permissions error in the remote functions sample.

**Pull request merged**